Automattic/jetpack
# Jetpack CLI: Add ability to generate plugin using a function starter-plugin as a template

Closes #23441
#### Changes proposed in this Pull Request:



This PR creates the Jetpack Starter Plugin and adds a new option to the generate plugin CLI command to build a new plugin using it as a template.

#### Jetpack product discussion


N/A

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Run `jetpack generate` CLI command
* Say you are generating a plugin
* When asked, choose the option to use the Starter plugin as a template
* When done, build the newly generated plugin with `jetpack build plugins/your-new-plugin`
* Test the plugin
* Verify that all names, slugs, class names and constants were renamed to match your new plugin name
* Test it with different plugin names, containing `-`, `_` and `.` in their names

Known issues
. [SOLVED] There's a hardcoded `autoloader suffix` in composer.json
. [SOLVED} When trying to commit, I got a babel error saying I needed to add a resolver for `jsx`, even though it is building successfully
. in `searchReplaceInFolder` there's a recursive call to itself that is not async. Could it generate problems?
. what else?
